server/index.js

```diff
@@ -11,17 +31,53 @@ app.use(bodyParser.json())
 
 const privateKey = '0x3daa79a26454a5528a3523f9e6345efdbd636e63f8c24a835204e6ccb5c88f9e'
 
+const exchangePublicToken = async token => {
+  const tokenResponse = await plaidClient.exchangePublicToken(token)
+  if (tokenResponse.status_code !== 200) {
+    logger.error({ status: tokenResponse.status_code }, 'Error exchanging public token')
+    throw Error(`[exchangePublicToken] ${tokenResponse.error_code}: {tokenResponse.error_message}`)
+  }
+  return tokenResponse
+}
+
+const getBankAccountData = async accessToken => {
+  const response = await plaidClient.getAuth(accessToken)
+  if (response.status_code !== 200) {
+    logger.error({ status: response.status_code }, 'Error getting bank account')
+    throw Error(`[getBankAccountData] ${response.error_code}: {response.error_message}`)
+  }
+  return response
+}
+
 app.post('/api/get-tx-data', (req, res) => {
-  const { ethAccount } = req.body
-  const bankAccount = 'MyAccount-' + Math.floor(1000 * Math.random())
+  const { ethAccount, token } = req.body
+
+  logger.info({ ethAccount }, 'Getting bank accounts')
+
+  return exchangePublicToken(token)
+    .then(tokenResponse => {
+      const accessToken = tokenResponse.access_token
+      return getBankAccountData(accessToken)
+    })
+    .then(accountData => {
+      logger.info({ accountData }, 'Got account data')
+      const { numbers } = accountData
+      const bankAccount = numbers[0].account
+
+      logger.info({ bankAccount }, 'Got bank account')
 
-  const hash = web3.utils.sha3(ethAccount + Buffer.from(bankAccount).toString('hex'))
-  const { v, r, s } = web3.eth.accounts.sign(hash, privateKey)
+      const hash = web3.utils.sha3(ethAccount + Buffer.from(bankAccount).toString('hex'))
+      const { v, r, s } = web3.eth.accounts.sign(hash, privateKey)
```

Comment:
The private key is hardcoded right now, but we should obtain it from an environment variable.